incompatible C-linkage specification

**Describe the bug**
On macOS, it seems like LLVM is spitting out a lot of warnings as vsg/core/Version.h is specifying an incompatible C linkage.

```
/Users/bret/Workspace/private/VulkanSceneGraph/cmake-build-relwithdebinfo/include/vsg/core/Version.h:40:43: warning: 'vsgGetVersion' has C-linkage specified, but returns user-defined type 'struct VsgVersion' which is incompatible with C [-Wreturn-type-c-linkage]
    extern VSG_DECLSPEC struct VsgVersion vsgGetVersion();
                                          ^
1 warning generated.
```

This gets generated almost for every .o file that is produced.

--- a/src/vsg/core/Version.cpp
+++ b/src/vsg/core/Version.cpp
@@ -17,7 +17,7 @@ extern "C"
 
     VsgVersion vsgGetVersion()
     {
-        VsgVersion version;
+        VsgVersion version{};
         version.major = VSG_VERSION_MAJOR;
         version.minor = VSG_VERSION_MINOR;
         version.patch = VSG_VERSION_PATCH;

--- a/src/vsg/core/Version.h.in
+++ b/src/vsg/core/Version.h.in
@@ -31,10 +31,10 @@ extern "C"
 
     struct VsgVersion
     {
-        unsigned int major = 0;
-        unsigned int minor = 0;
-        unsigned int patch = 0;
-        unsigned int soversion = 0;
+        unsigned int major;
+        unsigned int minor;
+        unsigned int patch;
+        unsigned int soversion;
     };
 
     extern VSG_DECLSPEC struct VsgVersion vsgGetVersion();

